test/libsolidity/semanticTests/array/byte_array_storage_layout.sol

```diff
@@ -0,0 +1,36 @@
+contract c {
+    bytes data;
+    function test_short() public returns (uint256 r) {
+        for (uint8 i = 0; i < 15; i++) {
+            data.push(bytes1(i));
+        }
+        assembly {
+            r := sload(data_slot)
+        }
+    }
+
+    function test_long() public returns (uint256 r) {
+        for (uint8 i = 0; i < 33; i++) {
+            data.push(bytes1(i));
+        }
+        assembly {
+            r := sload(data_slot)
+        }
+    }
+
+    function test_pop() public returns (uint256 r) {
+        for (uint8 i = 0; i < 32; i++) {
+            data.push(bytes1(i));
+        }
+        data.pop();
+        assembly {
+            r := sload(data_slot)
+        }
+    }
+}
+// ====
+// compileViaYul: also
+// ----
+// test_short() -> 1780731860627700044960722568376587075150542249149356309979516913770823710
+// test_long() -> 97
+// test_pop() -> 159
```

Comment:
This looks weird - did you mean to reset the length of data at the start of each function?

Reply:
I added reset with inline assembly now, as `delete`, or resize is not yet implemented for sol->yul.